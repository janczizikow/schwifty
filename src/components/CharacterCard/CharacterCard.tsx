import React from 'react';
import {View, Image, Text, StyleSheet} from 'react-native';
import theme from '../../utils/theme';

interface Props {
  image: string;
  name: string;
  status: string;
}

class CharacterCard extends React.PureComponent<Props, {}> {
  static HEIGHT = 100;

  render() {
    const {image, name, status} = this.props;

    return (
      <View style={styles.root}>
        <Image
          style={styles.image}
          source={{uri: image}}
          resizeMode="contain"
        />
        <View style={styles.meta}>
          <Text style={styles.name} numberOfLines={2}>
            {name}
          </Text>
          <Text style={styles.status}>{status}</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  root: {
    margin: 10,
    flexDirection: 'row',
  },
  image: {
    height: 80,
    width: 80,
    backgroundColor: theme.colors.border,
  },
  meta: {
    flex: 1,
    marginHorizontal: 10,
  },
  name: {
    fontSize: 24,
    fontWeight: '600',
    color: theme.colors.dark,
  },
  status: {
    color: theme.colors.grey,
  },
});

export default CharacterCard;
