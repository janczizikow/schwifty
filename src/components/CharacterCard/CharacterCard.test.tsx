import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCard from './CharacterCard';

const fakeProps = {
  name: 'Rick Sanchez',
  status: 'alive',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
};

describe('<CharacterCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CharacterCard {...fakeProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
