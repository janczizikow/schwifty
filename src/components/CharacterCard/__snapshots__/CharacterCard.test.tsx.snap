// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      }
    }
    style={
      Object {
        "backgroundColor": "#e7e7f0",
        "height": 80,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "color": "#2a2a30",
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Rick Sanchez
    </Text>
    <Text
      style={
        Object {
          "color": "#828393",
        }
      }
    >
      alive
    </Text>
  </View>
</View>
`;
